backend/danswer/connectors/confluence/connector.py

```diff
@@ -286,17 +447,39 @@ def _fetch(start_ind: int, batch_size: int) -> Collection[dict[str, Any]]:
 
                 return view_pages
 
+        def _fetch_page(start_ind: int, batch_size: int) -> list[dict[str, Any]]:
+            if self.recursive_indexer:
```

Comment:
I would prefer to instantiate it here to (1) remove the duplication and (2) not instantiate it unless we are using the recursive approach